front/src/modules/study/components/AddOrEditStudyModal.tsx

```diff
@@ -155,6 +156,18 @@ export default function AddOrEditStudyModal({ editionMode, study }: Props) {
     }
   }, [isCreateStudyError, isPatchStudyError, isDeleteStudyError]);
 
+  function getMaxDate(
+    expected_end_date: string | number | undefined,
+    actual_end_date: string | number | undefined
+  ) {
+    if (expected_end_date && actual_end_date) {
+      return expected_end_date < actual_end_date ? expected_end_date : actual_end_date;
+    }
+    if (expected_end_date) {
+      return expected_end_date;
+    }
+    return actual_end_date;
+  }
```

Comment:
You could simplify the function like that. 
```suggestion
 function getEarliestDate(date1: string | null | undefined, date2: string | null | undefined) {
    const formatedDate1 = formatDateForInput(date1);
    const formatedDate2 = formatDateForInput(date2);
    if (formatedDate1 && formatedDate2) {
      return formatedDate1 < formatedDate2 ? formatedDate1 : formatedDate2;
    }
    return formatedDate1 || formatedDate2;
  }
```

Reply:
Ty it's prettier ! Fixed